crates/agglayer-config/src/storage/backup.rs

```diff
@@ -0,0 +1,45 @@
+use std::path::PathBuf;
+
+use serde::{Deserialize, Serialize};
+
+/// Configuration for Storage backups.
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub enum BackupConfig {
+    /// Backups are disabled.
+    #[default]
+    Disabled,
+
+    /// Backups are enabled.
+    Enabled {
```

Comment:
Would it make sense to add `#[serde(untagged)]` here?
```suggestion
    /// Backups are enabled.
    #[serde(untagged)]
    Enabled {
```

Not sure if that would play nicely with toml serializer but it would avoid specifying the annoying `enabled` key in the config file.